authors/apps/authentication/tests/test_users.py

```diff
@@ -0,0 +1,141 @@
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+class TestUserLogin(BaseTest):
+    """
+    A class to test Login API endpoint.
+    Test user login credentials
+    """
+    def test_login_with_correct_credentials(self):
+        """
+        Test user login with the correct credentials
+        """
+        response = self.login_user("adam@gmail.com", "@Us3r.com")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_login_with_incorrect_credentials(self):
+        """
+        Test user login with the incorrect credentials
+        """
+        response = self.login_user("dam@gmail.com", "@Us3r.com")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+class TestUserUpdate(BaseTest):
+    """
+    A class to test Update API endpoint.
+    Test user update credentials
+    """
+    def test_update_with_user_login(self):
+        """
+        Test user update with the correct credentials
+        """
+        self.is_authenticated("adam@gmail.com", "@Us3r.com")
+        response = self.update_user("admin1", "adam@gmail.com", "@Us3r.com")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_update_with_user_logged_out(self):
+        """
+        Test user update with user not logged in
+        """
+        response = self.update_user("admin1", "adam@gmail.com", "@Us3r.com")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_update_with_user_details_exist(self):
+        """
+        Test user update with user details that exist
+        """
+        self.is_authenticated("admin@authors.com", "adm123Pass!!")
+        response = self.update_user("admin1", "adam@gmail.com", "@Us3r.com")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_update_with_invalid_value(self):
+        """
+        Test user update with invalid credentials
+        """
+        self.is_authenticated("admin@authors.com", "adm123Pass!!")
+        response = self.update_user("admin1", "adamgmail.com", "@Us3r.com")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        response = self.update_user("evegirl", "eve@gmail.com", "@Us")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+class TestGetUser(BaseTest):
+    """
+    A class to test Login API endpoint.
+    Test user login credentials
+    """
+    def test_with_logged_in_user(self):
+        """
+        Test get user with the correct login credentials
+        """
+        self.is_authenticated("adam@gmail.com", "@Us3r.com")
+        response = self.get_user()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_with_logged_out_user(self):
+        """
+        Test get user without login credentials
+        """
+        response = self.get_user()
```

Comment:
@a-braham please add a blank line at the end of the file to conform to pep8 standards